angular-workspace/projects/ni/nimble-angular/pipes/format-number-decimal.pipe.ts

```diff
@@ -0,0 +1,57 @@
+import { Inject, LOCALE_ID, Pipe, PipeTransform } from '@angular/core';
+import { DecimalUnitFormat } from '@ni/nimble-components/dist/esm/utilities/unit-format/decimal-unit-format';
+import { passthroughUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/passthrough-unit-scale';
+import { byteUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-unit-scale';
+import { byte1024UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-1024-unit-scale';
+import { voltUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/volt-unit-scale';
+import type { UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/unit-scale';
+
+export { byteUnitScale, byte1024UnitScale, voltUnitScale };
+
+/**
+ * A pipe that transforms a decimal number into a string with optional unit, formatted like "1.2 kB"
+ */
+@Pipe({
+    name: 'formatNumberDecimal',
+    standalone: true
+})
+export class FormatNumberDecimalPipe implements PipeTransform {
+    /**
+     * @internal
+     */
+    public decimalUnitFormat?: DecimalUnitFormat;
+    private minimumFractionDigits?: number;
+    private maximumFractionDigits?: number;
+    private unitScale?: UnitScale;
+
+    public constructor(@Inject(LOCALE_ID) private readonly locale: string) {}
+
+    public transform(value: number, {
```

Comment:
Most of Angular's pipes (e.g. number & date) have the options as separate parameters instead of one configuration object. Was it intentional to do something different here? Can you make sure that this works as expected (e.g. make sure that you can define the options only in the template, assuming you aren't setting a unit scale)?

Reply:
I did intentionally use a different pattern, because I feel it provides a better experience. E.g. if you only want to provide the units, it would be annoying to have to also provide the `minimumFractionDigits` and `maximumFractionDigits` just because of parameter order. This is especially true if I change the API to match the table column, like you suggested, because in that case, a client would have to provide both `digits` and `maximumFractionDigits`, which are mutually exclusive. If there were only two optional parameters, I would probably favor the typical (non-object) way of passing them instead. If the team feels we should not use an object parameter, I can change this.

> Can you make sure that this works as expected (e.g. make sure that you can define the options only in the template, assuming you aren't setting a unit scale)?

I don't follow. Are you asking me to create a test case that uses the pipe in a template, like this?
```HTML
<div>{{ values | formatNumberDecimal:{ maximumFractionDigits: 4 } }}</div>
```
What do you mean by "assuming you aren't setting a unit scale?" The unit scale can (only) be set through the pipe parameters in the template, too.

Reply:
> Are you asking me to create a test case that uses the pipe in a template, like this?

I'm not necessarily asking for a unit test case (though I'm not against it if you want to write one), but I just wanted to double check that we were certain that doing something like you showed of putting the object literal in the Angular template worked correctly. For example,
- Does the code snippet you showed compile?
- Does it correctly fail to compile if you set the wrong time? (e.g. `values | formatNumberDecimal:{ maximumFractionDigits: "not-a-numeric-value"}`

Part of the reason I asked is because [the only example I could find of something similar in Angular's docs](https://angular.io/api/common/I18nPluralPipe#example) creates the object in TS and then uses it in the template. I wanted to make sure that wasn't a requirement because it seems like that might be a pain for anyone to apply even the most basic configuration.

>What do you mean by "assuming you aren't setting a unit scale?" The unit scale can (only) be set through the pipe parameters in the template, too.

When you use a unit scale, that unit scale object will need to be created in the TS and then used in the template. I was trying to ask whether or not creating the configuration object without a unit scale could be done completely within the template (no code related to the pipe in the TS at all).

Reply:
>  (though I'm not against it if you want to write one)

Wrote one.

> Does it correctly fail to compile if you set the wrong type?

It doesn't fail to compile when in a test component in my test file, but it does fail to compile when in our example app's component template:
![image](https://github.com/ni/nimble/assets/7282195/2dee902e-6822-4aae-b29e-a39c614cd155)
I don't see a meaningful difference in the tsconfig files for the example and the tests, but I'm going to assume we're good as long as we're getting the error in the app.